app/controllers/api/restful_controller.rb

```diff
@@ -18,7 +20,8 @@ def destroy_action
   private
 
   def current_user
-    super || LoggedOutUser.new
+    token_user = User.find(doorkeeper_token.resource_owner_id) if doorkeeper_token
```

Comment:
Not quite sure I follow; this current_user method is used throughout the API (and I believe it's the only way we reference the current user anywhere in the API), so this should cover all of our API actions.

I would say caching the doorkeeper user so we don't have to hit the DB every time we reference current_user might be a good call though;

```
def current_user
  super || token_user || LoggedOutUser.new
end

private

def token_user
  @token_user ||= User.find(doorkeeper_token.resource_owner_id) if doorkeeper_token
end
```